pennylane/tape/tape.py

```diff
@@ -1149,21 +1150,23 @@ def func(a):
             >>> tape.numeric_type
             complex
         """
-        measurement_types = set(meas.return_type for meas in self._measurements)
+        measurement_types = {meas.return_type for meas in self._measurements}
         if len(measurement_types) > 1:
             raise TapeError(
                 "Getting the numeric type of a tape that contains multiple types of measurements is unsupported."
             )
 
         if list(measurement_types)[0] == qml.measurements.Sample:
-
-            for observable in self._measurements:
-                # Note: if one of the sample measurements contains outputs that
-                # are real, then the entire result will be real
-                if observable.numeric_type is float:
-                    return observable.numeric_type
-
-            return int
+            # Note: if one of the sample measurements contains outputs that
+            # are real, then the entire result will be real
+            return next(
+                (
+                    observable.numeric_type
+                    for observable in self._measurements
+                    if observable.numeric_type is float
+                ),
+                int,
+            )
```

Comment:
Neat! Is this much faster? Because it seems a bit less readable to me personally :sweat_smile:

Reply:
I don't think it is faster, but next is more "Pythonic". This change was recommended by a VS Code plugin called [Sourcery](https://sourcery.ai/).

But you are right, this makes the code less readable, so I will revert this change.

Reply:
Yeah, I definitely see how this it more "pythonic", maybe we will call this more readable in a few years :D